### PR TITLE
feat: get & list proposal

### PR DESCRIPTION
**Pull request by dayuy**
#26

### CODE REVIEW

File: pkg/utils/tools.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package utils
+
+import "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+// GetNestedString returns the string value of a nested field.
+// Returns "" if value is not found or not a string.
+func GetNestedString(obj map[string]interface{}, fields ...string) string {
```
Comment (bjwswang) [inline pkg/utils/tools.go:7]: add unit tests with https://codeium.com/ 

I recommend you use vscode plugin

File: pkg/proposal/get.go [hunk 2]
```diff
@@ -17,16 +17,115 @@ limitations under the License.
...
 
 func NewProposalGetCmd(option common.Options) *cobra.Command {
+	defaultPrintFlag := get.NewGetPrintFlags()
 	cmd := &cobra.Command{
-		Use: "",
-		RunE: func(cmd *cobra.Command, args []string) error {
-			return nil
+		Use: "proposal [NAME]",
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg, err := clientcmd.BuildConfigFromKubeconfigGetter("", common.InKubeGetter)
```
Comment (Abirdcfly) [inline pkg/proposal/get.go:44]: Similar logic appears multiple times, regardless of the resources being obtained. 
Can we abstract the similar parts into a function?

File: pkg/proposal/get.go [hunk 3]
```diff
@@ -17,16 +17,115 @@ limitations under the License.
...
+				TypeMeta: v1.TypeMeta{
+					Kind:       "List",
+					APIVersion: "v1",
+				},
+				ListMeta: v1.ListMeta{},
+			}
+			var obj runtime.Object
+			if len(args) == 0 {
+				username := viper.GetString("auth.username")
+				organizations, err := cli.Resource(schema.GroupVersionResource{Group: common.IBPGroup, Version: common.IBPVersion, Resource: common.OrganizationResource}).List(context.TODO(), v1.ListOptions{
```
Comment (bjwswang) [inline pkg/proposal/get.go:65]: please extract this selection as a common function in `pkg/org`, so we can reuse this func in other commands

File: pkg/utils/tools_test.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		},
+		{
+			name:     "empty string",
+			input:    []string{"a", "b", ""},
+			expected: []string{"a", "b"},
+		},
+		{
+			name:     "empty slice",
+			input:    []string{},
+			expected: []string{},
```
Comment (0xff-dev) [inline pkg/utils/tools_test.go:83]: Add a set of test cases:
`a , b, a, ""`

File: pkg/common/ibp.go [hunk 5]
```diff
@@ -85,6 +89,18 @@ func InKubeGetter() (*clientcmdapi.Config, error) {
 	}, nil
 }
 
+func GetDynamicClient() (*dynamic.DynamicClient, error) {
```
Comment (0xff-dev) [inline pkg/common/ibp.go:92]:
```go
func GetDynamicClient() (dynamic.Interface, error) {
}
```

`dynamic.Interface`

File: pkg/org/get.go [hunk 6]
```diff
@@ -119,3 +111,16 @@ func NewOrgGetCmd(option common.Options) *cobra.Command {
 
 	return cmd
 }
+
+// ListOrganizations returns a list of organizations filtered by labelSelector and fieldSelector.
+// Return error if any error occurs
+func ListOrganizations(cli *dynamic.DynamicClient, labelSelector string, fieldSelector string) (*unstructured.UnstructuredList, error) {
```
Comment (0xff-dev) [inline pkg/org/get.go:117]:
```go
func ListOrganizations(cli dynamic.Interface, labelSelector string, fieldSelector string) (*unstructured.UnstructuredList, error)
```

**Review by 0xff-dev — APPROVED** [review]

**Review by bjwswang — APPROVED** [review]

**Review by Abirdcfly — APPROVED** [review]

File: pkg/common/ibp.go [hunk 7]
```diff
@@ -85,6 +89,18 @@ func InKubeGetter() (*clientcmdapi.Config, error) {
 	}, nil
 }
 
+func GetDynamicClient() (dynamic.Interface, error) {
```
Comment (Abirdcfly) [inline pkg/common/ibp.go:92]: Nit: This function should be reused in other places as well.

**Review by Abirdcfly — APPROVED** [review]

**Pull request merged**